docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+
+### Add support for daily note templates
+
+With the above features implemented, making daily notes use templates is simple.
+
+We create a new `foam.defaultDailyNoteTemplate` setting which is the template that `Foam: Open Daily Note` will always use (defaults to `null`)
+
+Both `Foam: Open Daily Note` and `Foam: Create New Note` can share all of the implementation code, with the only differences being the hotkeys used and the default template setting used (`foam.defaultDailyNoteTemplate` vs. `foam.defaultNoteTemplate`).
+
+Example daily note template (again using the example syntax of the foam-specific frontmatter block):
+
+```markdown
+<!-- The below front-matter block is for foam-specific template settings -->
+---
+<!-- The default filepath to use when using this template -->
+<!-- Relative paths are relative to the workspace, absolute paths are absolute -->
+<!-- Note that you can include VSCode snippet variables to template the path -->
+filepath: `journal/${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}.md`
+---
+
+<!-- The actual contents of the template begin after the `---` thematic break immediately below this line-->
+---
+# ${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}
+```
+
+Since there is no use of the `${title}` variable, opening the daily note behaves exactly as it does today and automatically opens the note with no further user interaction.
+### Eliminate all `foam.openDailyNote` settings
+
+Now that all of the functionality of the `foam.openDailyNote` settings have been obviated, these settings can be removed:
+
+* `foam.openDailyNote.directory`, `foam.openDailyNote.filenameFormat`, and `foam.openDailyNote.fileExtension` can be specified in the `filepath` metadata of the daily note template.
+* `foam.openDailyNote.titleFormat` has been replaced by the ability to fully template the daily note, including the title.
+
+## Summary: resulting behaviour
+
+### `Foam: Create New Note`
+
+A new command optimized for speedy creation of new notes. This will become the default way to create new notes. In its fastest form, it simply opens the new note with no further user interaction.
+
+### `Foam: Open Daily Note`
+
+Simplified since it no longer has its custom settings, and re-uses all the same implementation code as `Foam: Create New Note`.
+Templates can now be used with daily notes.
+
+### Navigating to missing wikilinks
+
+Now creates the new notes using the default template. Re-uses all the same implementation code as `Foam: Create New Note`
```

Comment:
and uses the wikilink as value for `${title}`

Reply:
Added a mention of this.